Win package

Pat-Hof opened the pull request:
Adds 2 options to the Win_package module:

- display_name
- display_version

It permit either the display_name of a software or the display_name and the display_version to check its installation state.

How its works : 

If the name and optionally the version of a software are present in the x86 or x64 Uninstall registry trees, its state is installed. if found, it gets the uninstall string for deinstallation.

The display_name and optionally the display_version can be found in the Programs and Features Panel 

Examples :
```yaml
- name: Install 7zip and use the display_name for the installation check found in
Programs and Features Panel
  win_package:
    path: C:\temp\7zip.exe
    display_name: 7-Zip 16.04 (x64 edition)

- name: Install Zotero use the displayname and displayversion
   win_package:
     path:  C:\temp\Zotero-5.0.52_setup.exe
     arguments: /S
     display_name: Zotero
     display_version: 5.0.52
```

##### ISSUE TYPE
 
- Feature Pull Request

##### COMPONENT NAME

win_package

##### ANSIBLE VERSION
ansible 2.5.5
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.15 (default, May  1 2018, 05:55:50) [GCC 7.3.0]

##### ADDITIONAL INFORMATION
Tested on Windows 10 platform and for the error conditions as well.

Yannig committed:
Fix with_ini example and unittest

* Fix example in ini.py
* Fix unittest in test_ini.py to pass CI as latest ansible returns list in
  different order. To prevent such issues in future results are sorted
* PEP8 E501 styling improvements

Co-Authored-By: Sergii Golovatiuk <sgolovat@redhat.com>

alikins committed:
Bundle a copy of github.com/nir0s/distro/distro.py

Since the 'platform.dist()' and 'platform.linux_distribution()'
methods will be removed from future versions of python, this
provides an alternative to replace ansibles use of those
methods.

lib/ansible/module_utils/distro.py is a copy of
https://github.com/nir0s/distro/blob/master/distro.py

This module is originally from https://github.com/nir0s/distro
and is license under the Apache License, Version 2.0.

alikins committed:
WIP: add tests to compare to platform.*dist

Can be removed once everything is ported.

alikins committed:
Changes to bundled distro to be Python-2.6 compatible

* Port bundled distro to use optparse instead of argparse (py2.6)
* Use an absolute import to satisfy the current import testing harness
* Port from subprocess.check_output to subprocess.Popen.communicate() (py2.6)
* Add license location

The changes have been proposed upstream here:
https://github.com/nir0s/distro/pull/232

Upstream is contemplating a branch where everyone wanting python-2.6
support can collaborate without it becoming part of the regularly
supported releases.

alikins committed:
Skip sanity tests that don't apply to bundled code

* add distro to pep8 skip tests
* Skip no-assert test for distro
* Add bundled distro to the empty-init skip list

piffey committed:
Move verify up so approle and other methods work as intended.

pathof-unistra committed:
add check if software is installed based on name and version from Programs and Features panel

pathof-unistra committed:
modify latest win_package.py with new parameters

pathof-unistra committed:
allow display_name set and display_version unset if display_name contain software version. force display_name to be set if display_version is set

pathof-unistra committed:
display_version is now optional

pathof-unistra committed:
enable display_name option when deinstalling software without path info

pathof-unistra committed:
add examples with display_name and display_version options

pathof-unistra committed:
documentation improvements

pathof-unistra committed:
comment updates and change condition when the path is not an MSI or the path is an MSI but not local : product_id or display_name required

pathof-unistra committed:
comment updates and change condition; restore test condition when to get UninstallString from registry

pathof-unistra committed:
yml documentation fixes

pathof-unistra committed:
yml documentation fixes two

pathof-unistra committed:
add version_added for new option for new options

mattclay commented:
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/73315/22/tests

dagwieers commented:
Beware that [the deadline](https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/roadmap/ROADMAP_2_7.rst#release-schedule) for getting new features/modules accepted in Ansible v2.7 is nearing, it is set to either 2018-08-23 or 2018-08-30. If you are blocked, or you need feedback, please discuss on IRC channel #ansible-windows or add a comment to the [Windows Working Group meeting agenda](https://github.com/ansible/community/issues/294) to get it resolved.

jborean93 commented on lib/ansible/modules/windows/win_package.ps1:
```diff
@@ -221,12 +227,21 @@ Function Get-ProgramMetadata($state, $path, $product_id, $credential, $creates_p
         }
     }
 
+    if ($display_name -ne $null) {
```
You may as well set this when `$metadata` is created, the value will still be `$null` when `$display_name/version` is also null

Pat-Hof replied:
Hi!
Sorry for the long silence
Changes done

jborean93 commented on lib/ansible/modules/windows/win_package.ps1:
```diff
@@ -221,12 +227,21 @@ Function Get-ProgramMetadata($state, $path, $product_id, $credential, $creates_p
+        $metadata.display_name = $display_name
+    }
+
+    if ($display_version -ne $null) {
```
Same as above, no need for an if statement just set display_version in `$metadata` to `$display_version`

Pat-Hof replied:
Changes done

jborean93 commented on lib/ansible/modules/windows/win_package.ps1:
```diff
@@ -248,11 +263,65 @@ Function Get-ProgramMetadata($state, $path, $product_id, $credential, $creates_p
+                }
+            }
+        }
+        foreach ($product_id in $uninstall_array_wow64) {
```
Just join the 2 arrays together and run the loop on all the entries.

Pat-Hof replied:
done

jborean93 reviewed (changes requested):
It would be great if you could add some integration tests, feel free to reach out if you are unsure on how to do this.

abadger committed:
Enable the bare-except pylint and pep8 checking

abadger committed:
Update bare exceptions to specify Exception.

This will keep us from accidentally catching program-exiting exceptions
like KeyboardInterupt and SystemExit.

abadger committed:
Ensure that current uses of BaseException are required

* In some cases, it appears that Exception should have been used instead
  as there's no need to catch sys.exit KeyboardInterrupt and similar.
* In a few cases, it appears that BaseException is used because
  a library we depend on calls sys.exit() contrary to good coding
  design.  Comment those so that we know that those have been audited
  and found to be correct and change to use (Exception, SystemExit)
  instead.

abadger committed:
Fix validate-modules to not complain about sys.exit in comments

IPvSean committed:
adding example for route53_facts using start_record_name

this example shows two things not shown here already->
- dynamically looking up hosted zone id using the route53_zone module (vs knowing the ID we can use the name)
- showing an example of start_record_name which takes an entire record, not just a partial name

s-hertel committed:
Lowercase task name and format variable with spaces

abadger committed:
Port from plaform.dist to ansible.module_utils.distro.linux_distribution

ci_complete

cma0 committed:
itential iap_token module (#46773)

richlv committed:
Update rhsm_repository.py (#49480)

typo fixes

beeankha committed:
Making changes to the Community Guide index (#49670)

badstreff committed:
added support for icmp-blocks and icmp-block-inversion (#45705)

RBakkum committed:
Fix example with spacing error in hostname (#49812)

fix na_ontap_command module examples

dagwieers committed:
Docs: Add a "seealso" section to the module docs (#45949)

* Docs: Add a separate  "seealso" section to the module docs
to list related modules and/or related references. This clears up the notes
section for things that are actual notes.

So you can add a section in your module documentation and four types of
references are possible.

    seealso:

    # Reference by module name
    - module: aci_tenant

    # Reference by module name, including description
    - module: aci_tenant
      description: ACI module to create tenants on a Cisco ACI fabric.

    # Reference by rST documentation anchor
    - ref: aci_guide
      description: Detailed information on how to manage your ACI infrastructure using Ansible.

    # Reference by Internet resource
    - name: APIC Management Information Model reference
      description: Complete reference of the APIC object model.
      link: https://developer.cisco.com/docs/apic-mim-ref/

This PR also includes:

- Implements ansible-doc support
- Implements schema support for the seealso options
- Updates to the development documentation
- Rename filter convert_symbols_to_format to rst_ify, cfr the existing html_ify and tty_ify filters
  - This makes the existing template a lot easier to read and fixes the confusion I had myself rereading the template (again).
- We fixed the possible suboption types (which was limited to 'bool' only)

* Use latest stable instead of devel docs

acozine committed:
link from API stub page to module_utils page (#49831)

rhydian76 committed:
Create Windows facts for ansible_virtualization_role and ansible_virt… (#49775)

* Create Windows facts for ansible_virtualization_role and ansible_virtualization_type

* Create Windows facts for ansible_virtualization_role and ansible_virtualization_type
Updated formatting

* Removed executable flag on setup.ps1

* Changed Get-WMIObject to Get-CimInstance as Get-WMIObject has been deprecated and fails testing.

* Changed new variables to snake_case and also changed Get-CimInstance to Get-LazyCimInstance

* removed -class from Get-LazyCimInstance call as failed test.

gstorme committed:
replace math round with ceiling for ansible_memtotal_mb + add new variables to display swapfile config (#49611)

* replace math round with ceiling to report the actually installed RAM for ansible_memtotal_mb

* add new variable ansible_memtotal to display TotalPhysicalMemory in bytes
add new variable ansible_swap_min to display initial pagefile size
add new variable ansible_swap_max to display maximum pagefile size
removed the variable ansible_swaptotal_mb

* output ansible_swap_min & ansible_swap_max value in bytes

* re-add the ansible_swaptotal_mb fact and fix the conversion of the TotalSwapSpaceSize value from kilobytes to MB

* indentation fix, replace tab with spaces

jborean93 committed:
win become: refactor and add support for passwordless become (#48082)

* win become: refactor and add support for passwordless become

* make tests more stable

* fix up dep message for Load-CommandUtils

* Add further check for System impersonation token

* re-add support for become with accounts that have no password

* doc fixes and slight code improvements

* fix doc sanity issue

ssahani committed:
nmcli: Introduce SIT Tunnel (#45937)

This work enables to add sit tunnel via nmcli module

Signed-off-by: Susant Sahani susant@redhat.com

SUMMARY
ISSUE TYPE
Feature Pull Request
COMPONENT NAME
nmcli

ANSIBLE VERSION
2.8
ADDITIONAL INFORMATION
  - nmcli:
       state: present
       type: sit
       conn_name: sit_test1
       autoconnect: yes
       ip_tunnel_dev: enp0s8
       ip_tunnel_local: 192.168.1.2
       ip_tunnel_remote: 192.168.1.5

FragmentedPacket committed:
Netbox Module: netbox_ip_address (#48424)

* Tested netbox_ip_address with several conditions and working as intended

greg-hellings committed:
Properly report install failures from dnf module (#49760)

Reporting of install failures was improperly guarded behind a list of
packages to install when the first element changed in a list was found.
This lead to the dnf module silently failing at times when it should not
fail.

Fixes issue 49759

jborean93 committed:
Revert "replace math round with ceiling for ansible_memtotal_mb + add new variables to display swapfile config (#49611)" (#49876)

This reverts commit b3ac5b637a4ff6259c82b659517ab04f5c0b2f11.

Akasurde committed:
Update unit tests for configmanager

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

trishnaguha committed:
add host key as alias for sysname neighbors facts (#49827)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

dagwieers committed:
win_updates Add information/example wrt. reboots (#49834)

This documents two common issues related to rebooting Windows systems.
- Updates can take a long time being applied during reboot
- Reboots can cause the WinRM service to start before system has settled

bcoca committed:
corrected docs on args

gundalow committed:
BOTMETA module_utils/ec2.py us aws (#49899)

dagwieers committed:
ACI: Add seealso fragments to module documentation (#49844)

jborean93 committed:
windows dev docs: fix a few issues and add missing info (#49836)

nerzhul committed:
postgresql_schema: permit to drop cascade schema (#49469)

* postgresql_schema: permit to drop cascade schema

* Doc fix

* typo fix

jborean93 committed:
ansible-test: do not upgrade homebrew to speed up tests (#49914)

jborean93 committed:
Windows Privileges - moved util code to it's own C# util (#48897)

* Windows Privileges - moved util code to it's own C# util

* Rename Enabler class to PrivilegeEnabler to remove ambiguity

* rename Utils to PrivilegeUtil

* fix missing util name changes

Akasurde committed:
Review comments

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

anasbadaha committed:
Get Capabilities Support (#49880)

Signed-off-by: Anas Badaha <anasb@mellanox.com>

gdpak committed:
ignore connection timeout out within parenthesis (#48413)

yuwzho committed:
support overprovision in azure_rm_vmss (#49806)

Akasurde committed:
Misc typo fixes (#49816)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
Return correct error message to user

botocore.exceptions.ClientError does not have message, which
fails fail_json.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

acozine committed:
removes space from example of L(link) syntax (#49908)

ruzickap committed:
with_items should not be used any more - using loop instead (#47652)

ruzickap committed:
with_items should not be used any more - using loop instead (#47364)

ruzickap committed:
with_items is obsolete - using loop instead (#47163)

peppos committed:
issue with activation_key and proxy on subscription-manager (#42624)

* Update redhat_subscription.py

I need to use a proxy server/user/password also if I user an activation_key

* remove spaces

akshay196 committed:
Support for cache_from parameter in docker_image module (#49787)

* Support for cache_from parameter in docker_image module

Fixes #21704

Signed-off-by: Akshay Gaikwad <akgaikwad001@gmail.com>

* Minimum docker version check for cache_from option

Signed-off-by: Akshay Gaikwad <akgaikwad001@gmail.com>

* Add changelog fragment

Signed-off-by: Akshay Gaikwad <akgaikwad001@gmail.com>

* cache_from fix when docker-py is older

Signed-off-by: Akshay Gaikwad <akgaikwad001@gmail.com>

DBendit committed:
[docker_volume] Checking option minimal versions (#49839)

Fixes #38833

DBendit committed:
[docker_container] Failing on non-string env values (#49843)

* [docker_container] Failing on non-string env values

Fixes #49802

* Clarify failure message

Co-Authored-By: DBendit <David@ibendit.com>

* Fixup from review

maxamillion committed:
Fix yum/dnf handling of URIs that don't end in .rpm (#49912)

Fixes #49727

Signed-off-by: Adam Miller <admiller@redhat.com>

maxamillion committed:
add changelog for pr #49912 (#49962)

Signed-off-by: Adam Miller <admiller@redhat.com>

sivel committed:
Remove erroneous extra-vars example, and a few really old versionadded indicators. Fixes #38415 (#49955)

mkrizek committed:
docs: remove not implemented password_hash feature (#49945)

dagwieers committed:
Add seealso section to Windows and related modules (#49845)

* Add seealso section to Windows and related modules

* Fixes after review

Akasurde committed:
fortimanager: Correct example documentation (#49925)

Fixes: #49813

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
Ad-hoc command: fix 'any' call in play_ds (#49852)

* Ad-hoc command: fix 'any' call in play_ds
* Unit test for ad-hoc cli
* Review comments

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
AWS: make jittered_backoff API parameter configurable (#49086)

* AWS: make jittered_backoff API parameter configurable
* Review comments
* minor doc changes

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

anasbadaha committed:
Issue/38837 (#49873)

* Fix Issue #38837

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Fix Issue #38837 2

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Enhance decumentation

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Fix Failures in Shippable

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Fix Version Added

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Add version_added:2.8 in purge attributes

Signed-off-by: Anas Badaha <anasb@mellanox.com>

felixfontein committed:
Fix stacktrace when fail_on_error is False and a file cannot be parsed. (#49987)

Commit:
support lro in azure_rm_resource (#49919)

anasbadaha committed:
Issue: 46475 Fix onyx magp module for supporting new json format (#49417)

Signed-off-by: Anas Badaha <anasb@mellanox.com>

anasbadaha committed:
Fixes #46410 Onyx show facts module does not work for onyx version 3.6.6000 and higher (#49414)

Signed-off-by: Anas Badaha <anasb@mellanox.com>

f-bor committed:
new network module: edgeswitch_vlan (#48041)

* initial commit

* better commands generation

gdpak committed:
network_cli - ansible_command_timeout not working as expected (#49922)

* fix for command_timeout

Signed-off-by: Deepak Agrawal <deepacks@gmail.com>

* fix ci warning

Signed-off-by: Deepak Agrawal <deepacks@gmail.com>

* fix review comments

Signed-off-by: Deepak Agrawal <deepacks@gmail.com>

dagwieers committed:
Add types to docker module fragments (#50002)

Andersson007 committed:
New module - postgresql_idx (#49684)

* New module - postgresql_idx

* New module postgresql_idx: added returns block

* New module postgresql_idx: fixed documentation, typos and added idxname to fail_json

MatrixCrawler committed:
Sophos UTM - add module to handle utm ca host key certificate entries (#48927)

* - add module to handle utm ca host key certificate entries

* - fixed typos in doc

* Apply suggestions from code review

Co-Authored-By: MatrixCrawler <johannes.brunswicker@gmail.com>

Commit:
Docker inventory service/stack groups for docker swarm (#48021)

* Adding service and stack grouping to docker inventory

* Updating documentation

* Adding changelog fragment for docker.py inventory script

JeffReeves committed:
Redacted IP addresses under SSH_CLIENT & SSH_CONNECTION (#49991)

gundalow committed:
Gundalow no net docs updates (#50031)

mnecas committed:
add to ovirt snapshot download and upload disk (#46696)

* add to ovirt snapshot download and upload disk

* ovirt snapshot update docs

* ovirt snapshot add version_added to docs

* update ovirt snapshot docs

dagwieers committed:
Fix document references in modules (#49892)

* Docs: Fixes internal module reference syntax for seealso
* Updates anchors and links
* Updates seealso in the docs for module **win_chocolatey**.

dagwieers committed:
Add types to vmware REST client/VCA doc fragments (#50001)

dagwieers committed:
Clean up backup doc fragment (#49995)

dagwieers committed:
ACI: Add types to modules and  doc fragment (#49997)

* ACI: Add types to module doc fragment

* Add types to ACI modules too

dagwieers committed:
MSC: Add type to common module parameters (#49999)

dagwieers committed:
Meraki: Add types to common parameters (#49998)

dagwieers committed:
Add types to common url parameters (#50000)

Qalthos committed:
Move errors from import to connect (#50034)

Qalthos committed:
Don't fail if a remote_addr with a '/' hits ansible_connection (#49781)

* Fail if a remote_addr with a '/' hist ansible_connection

This is _probably_ a CIDR block, but anything with a slash will fail,
so no need to try to parse to make sure

* Locks are now per-socket_path.

Locks use the same value as socket_path. Locks are also cleaned up in
shutdown like sockets.

dagwieers committed:
Fix GitHub handle, use keachi (#50043)

* Fix GitHub handle, use keachi

* Add UTF8 name

mattclay committed:
Fix MySQL tests when running under Docker. (#50047)

Alexander198961 committed:
changed  if logic Type to boolean (#49628)

* changed Type to boolean

* fix after review

* created fragment

mattclay committed:
Add OverlayFS work-around for CloudStack tests.

sivel committed:
Fix reverse_inventory order to work on python3 (#49895)

sivel committed:
Allow tags to be templated from a variable (#49833)

* Allow tags to be templated from a variable. Fixes #49825

* Restore _load_tags to ensure we do csv tag splitting

* Add tests for csv tags and templated tags

* evaluate_tags doesn't need to accept strings, because _load_tags handles this

jlecour committed:
Lineinfile must not insert lines multiples times with insertbefore/insertafter (#49409)

* Change test suite to fit expected behaviour

This reverts some changes from ansible/ansible@723daf3
If a line is found in the file, exactly or via regexp matching, it must 
not be added again.
insertafter/insertbefore options are used only when a line is to be 
inserted, to specify where it must be added.

* Implement the change in behaviour mentioned in the previous commit

* Fix comment to reflect what the code does

* Set the correct return message.

In these cases, the lines are added, not replaced.

* Add a changelog

gstorme committed:
fix facts memtotal_mb rounding on VMWare and swaptotal_mb conversion from KB to MB (#49940)

dagwieers committed:
Add types to files fragment (#49996)

dagwieers committed:
PSRP: Add support for reconnection retries (#49772)

* PSRP: Add support for reconnection retries

This adds support for reconnection retries to overcome sudden Connection
Refused or other recoverable states. (Either because WinRM is (re)started,
or firewall changes make it briefly unavailable)

* Fixes from review comments

samccann committed:
link to KB article -certified modules (#50050)

abadger committed:
Remove dead code in ansible-inventory

This piece of code is only needed with ansible < 2.3.x (which is now EOL)

ganeshrn committed:
Add force_host and issu options support in junos_package module (#50018)

Fixes #48171
Fixes #33742

Add force_host and issu options support in junos_package module
*  Add force_host option to ignore warnings while installing package
*  Add issu option to enable issu feature from Ansible

rajaspachipulusu17 committed:
Pluribus Networks cpu class module with unit tests (#49847)

* Pluribus Networks cpu class module with unit tests
* Doc string fixes

rajaspachipulusu17 committed:
Pluribus Networks dscp map module with unit tests (#49849)

* Pluribus Networks dscp map module with unit tests
* Doc string fix

dagwieers committed:
ACI: Fix module parameters (#50051)

* Fix parameter type of private_key

* Fix parameters of various ACI modules

Commit:
fixing vm state change when vm created with attached disk (#50071)

* fixing vm state change when vm created with attached disk

* minor fixes

* fixing hanging indent

felixfontein committed:
openssl_privatekey: add ECC support (#49416)

* Add cryptography backend for openssl_privatekey.

* Adding ECC support.

No support for X25519 and X449, since they don't support serialization.

* Improve finterprint calculation to work with Python 3.

* Add fingerprint check.

* Fix typo.

* Use separate curve option for elliptic curves, and use type 'ECC'.

* Using curve names as defined in IANA registry.

* Bump minimal supported cryptography version. Older versions might work as well, but I couldn't test them.

* Improve documentation.

ckotte committed:
VMware: new module vmware_dvswitch_uplink_pg (#48846)

machacekondra committed:
Fix activate of the disks in ovirt_vm and ovirt_disk modules (#49762)

* ovirt_disk: Add activate parameter

* ovirt_vm: Fix waiting on lun disks

Signed-off-by: Ondra Machacek <omachace@redhat.com>

ckotte committed:
VMware: new module vmware_host_ad_auth (#47901)

ckotte committed:
Improve module vmware_local_role_manager (#47870)

* add check mode support
* improve output

rajaspachipulusu17 committed:
Pluribus networks admin session timeout module with unit tests (#50012)

* Pluribus networks admin session timeout module with unit tests
* Doc fix
* Removed unused imports
* Removed unwanted global variable

webknjaz committed:
Upgrade README to have 4k+ contributors listed (#50093)

Qalthos committed:
ios retry config if section filter fails (#49485)

* Attempt to work around devices that don't understand | section

* Fix case of no flags

dagwieers committed:
Sanity fixes in various modules (#50080)

xmadsen committed:
GetManagerNicInventory (#49961)

* Add GetManagerNicInventory command for Manager category, and tweak redfish_utils' get_nic_inventory to accommodate manager nic interfaces

* Remove get_manager_nic_inventory, since it is unnecessary while using get_manager in get_nic_inventory

* Rework get_nic_inventory() to take a resource_type as a string, which will be just the category parameter from redfish_facts, making it clearer

* remove extraneous blank line to conform with pep8

* Add GetManagerNicInventory example task to EXAMPLES docstring

bcoca committed:
added handlers_from (#49220)

* added handlers_from

fixes #46769

dagwieers committed:
Remove older version_added information from docs (#50097)

dagwieers committed:
plugin_formatter.py: Improve the output when processing docs (#46541)

* Improve the output when processing files

* Update docs/bin/plugin_formatter.py

Co-Authored-By: dagwieers <dag@wieers.com>

* Show progress indicator.

* Don't pp.pformat() huge structures when they aren't used anyway.

This saves ~10 seconds on my machine.

* Only show ASCII spinner if stdout is a TTY.

* Fix: E722 do not use bare 'except'

sivel committed:
Remove the use of mock from validate-modules (#50098)

* Remove the use of mock from validate-modules

* Monkeypatch AnsibleModule.__init__ not AnsibleModule

mscherer committed:
Do switch to apt if aptitude is not installed (#30738)

Trying to use the apt module on a freshly install 16.04 Ubuntu VM with
"ansible -m apt -a 'name=* state=latest' all" fail with this backtrace:

  Traceback (most recent call last):
    File \"/tmp/ansible_2inxygge/ansible_module_apt.py\", line 1055, in <module>
      main()
    File \"/tmp/ansible_2inxygge/ansible_module_apt.py\", line 998, in main
      upgrade(module, 'yes', force_yes, p['default_release'], use_apt_get, dpkg_options)
    File \"/tmp/ansible_2inxygge/ansible_module_apt.py\", line 759, in upgrade
      apt_cmd_path = m.get_bin_path(apt_cmd, required=True)
    File \"/tmp/ansible_2inxygge/ansible_modlib.zip/ansible/module_utils/basic.py\", line 2182, in get_bin_path
    File \"/usr/lib/python3.5/posixpath.py\", line 89, in join
      genericpath._check_arg_types('join', a, *p)
    File \"/usr/lib/python3.5/genericpath.py\", line 143, in _check_arg_types
      (funcname, s.__class__.__name__)) from None
    TypeError: join() argument must be str or bytes, not 'NoneType'

It seems that since aptitude is not installed on Xenial, so APTITUDE_CMD is None
and we later hit the bug in the upgrade function since it assume APTITUDE_CMD
is set.

michaeldg committed:
Small documentation update: Clarify daemon-reload always executing. (#49889)

* Small documentation update
* shortens and clarifies line

dagwieers committed:
no-smart-quotes: Skip files by path in code-smell (#45881)

* no-smart-quotes: Skip files by path in code-smell

This is to ensure the generated docs do not trigger code-smell issues on
contributor systems.

* Implement feedback from review

jborean93 committed:
win_lineinfile - fix malformed returned json (#50066)

mkrizek committed:
facts: correctly detect multiple ipv6 addresses per device (#50020)

Fixes #49473

mkrizek committed:
facts: detect FreeBSD KVM guests (#50030)

Fixes: #49158

abadger committed:
VALID_ACTIONS for cli subcommands will now be a frozenset (#50058)

dagwieers committed:
Convert to reduced list of known types (#50010)

abadger committed:
Fix return types

We're now stricter on naming the return types.  Replace string with str.

tdietrich513 committed:
Warn of standard behavior of 'default' key store (#49903)

* Warn of standard behavior of 'default' key store 

The default key store typically results in the key being stored as user. In instances where we're trying to install a cert to enable SSL binding in IIS, this will result in an unbind-able cert, as the IIS APPPOOL identity will not be able to read the private key, and presents a very cryptic error of 'A specified logon session does not exist'.


+label: docsite_pr

* fix trailing whitespace

* Use C(user) reference

* specify what's required for successful iis import

- store_location's default value is currently 'LocalMachine' but if this changes it could result in the documentation being inaccurate
- key_exportable is not required for a successful IIS import

* remove stray =

dang cat

Co-Authored-By: tdietrich513 <tom.dietrich@gmail.com>

abadger committed:
Make the timeout decorator raise an exception out of the function's scope (#49921)

* Revert "allow caller to deal with timeout (#49449)"

This reverts commit 63279823a77320b8cdf7f69d61af6eddaa5ebf33.

Flawed on many levels

* Adds poor API to a public function
* Papers over the fact that the public function is doing something bad
  by catching exceptions it cannot handle in the first place
* Papers over the real cause of the issue which is a bug in the timeout
  decorator
* Doesn't reraise properly
* Catches the wrong exception

Fixes #49824
Fixes #49817

* Make the timeout decorator properly raise an exception outside of the function's scope

signal handlers which raise exceptions will never work well because the
exception can be raised anywhere in the called code.  This leads to
exception race conditions where the exceptions could end up being
hanlded by unintended pieces of the called code.

The timeout decorator was using just that idiom.  It was especially bad
because the decorator syntactically occurs outside of the called code
but because of the signal handler, the exception was being raised inside
of the called code.

This change uses a thread instead of a signal to manage the timeout in
parallel to the execution of the decorated function.  Since raising of
the exception happens inside of the decorator, now, instead of inside of
a signal handler, the timeout exception is raised from outside of the
called code as expected which makes reasoning about where exceptions are
to be expected intuitive again.

Fixes #43884

* Add a common case test.

Adding an integration test driven from our unittests.  Most of the time
we'll timeout in run_command which is running things in a subprocess.
Create a test for that specific case in case anything funky comes up
between threading and execve.

* Don't use OSError-based TimeoutError as a base class

Unlike most standard exceptions, OSError has a specific parameter list
with specific meanings.  Instead follow the example of other stdlib
functions, concurrent.futures and multiprocessing and define a separate
TimeoutException.

* Add comment and docstring to point out that this is not hte Python3 TimeoutError

wojtek0806 committed:
added ASM policy server technology module (#50116)

* added ASM policy server technology module

* Update bigip_asm_policy_server_technology.py

Commit:
fixed crash in facts (#50120)

wojtek0806 committed:
added bigip_asm_policy_manage (#50115)

* added bigip_asm_policy_manage

* Update bigip_asm_policy_manage.py

wojtek0806 committed:
added asm endpoint facts (#50114)

* added asm endpoint facts

* Update bigip_device_facts.py

wojtek0806 committed:
changed oData filters (#50112)

* changed oData filters

* Update bigip_asm_policy_fetch.py

* Update bigip_asm_policy_import.py

wojtek0806 committed:
fixed an issue with parameter breaking idempotency (#50117)

* fixed an issue with parameter breaking idempotency

* Update bigip_pool_member.py

jborean93 committed:
win_copy - fix remote dir copy when it contains an empty dir (#50126)

mattclay committed:
Fix no metadata traceback in validate-modules.

ganeshrn committed:
Fix mandatory statement error for junos modules (#50074)

* Fix mandatory statement error for junos modules

Fixes #40267

*  Add error regex in junos terminal plugin to error out
   in case of commit fails

*  If commit fails add logic to discard changes before existing
   else next task will result in error

* Add integration test

* Minor update

lDA committed:
Fix incorrect examples with random filter (#50137)

mbork committed:
Update git.py (#50139)

* Update git.py

Mention `accept_hostkey` in the description of `ssh_opts`

+label: docsite_pr

* Small fix to formatting

soulofmischief committed:
Fix typo: or -> on (#50123)

"To create a disabled account or Linux systems" should be "To create a disabled account **on** Linux systems"

+label: docsite_pr

rajaspachipulusu17 committed:
Pluribus Networks admin service module with unit tests (#49854)

* Pluribus Networks admin service module with unit tests
* Documentation fix
* import module changes
* Removed unused imports
* Fix return type

rajaspachipulusu17 committed:
Pluribus Networks dhcp filter module with unit tests (#49848)

* Pluribus Networks dhcp filter module with unit tests
* Added type which was missing in doc
* Removed unwanted global variable
* Fix return type

Commit:
Added organization in the scm_credential get (#49884)

* Added organization in the scm_credential get

* Fallback looking for cred in project org

* Tests project with multi org credential

* Fixed CI issue

* Added changelog fragment

sn4kebite committed:
rabbitmq_binding: Fix using empty routing key (#48597)

* rabbitmq_binding: Fix using empty routing key

If the routing key is an empty string we need to use the special props
value ~ in the URL. Otherwise the last part of the URL will be empty,
which will instead be equivalent to the "list all bindings" API call
which always returns 200 OK (as long as the exchange and queue exists).

* rabbitmq_binding: Move routing_key test outside format

Test the routing_key value and set the props value outside the format
call. We leave the original routing_key untouched since its original
value is later needed in create().

wojtek0806 committed:
corrected valid type values in RETURN (#50148)

bcoca committed:
dont think all files are valid cause libs present (#50103)

verify_file was improperly always returning true if pyvimomi and requests libs were correct
  moved library checking to parse, avoid unneded errors unless the file is actually meant for
  this plugin

Qalthos committed:
Allow persistent connection plugins to queue messages back to ansible-connection (#49977)

* Connections can queue messages to be returned from ansible-connection

* Provide fallback for invalid display level

* Strip display from plugins

* Route messages through helper method to try to avoid improper appends

Commit:
Refactoring of cnos_interface module like what followed by other vendors (#49927)

* Refactoring of cnos_interface module like what followed by other vendors

* To remove cnos-interface from E326 validation ignore list

* Effect of default parameters impacted UT

bcoca committed:
Revert range change (#50155)

* Revert "Fix incorrect examples with random filter (#50137)"

This reverts commit 9a7dbd52139f22c9c0e5f3f127efb0f8956c6f6a.

The correction is incomplete and also based on a 'fix' that was supposed to have been reverted already

* Revert "Added `+1` to the `end` in `random` filter so that it was inclusive (#27215)"

This reverts commit ea2b89c7ae69424055652db30f49182d4a987bf7.

reverted fix as agreed at the time, but missed by maintainers.

AlanCoding committed:
update openshift inventory plugin docs (#49901)

mhoffmann1 committed:
VMware: Update example vmware_deploy_ovf (#49893)

* Update vmware_deploy_ovf.py
Provided more detailed example using additional, often used parameters
Co-Authored-By: mhoffmann1 <marcin.hoffmann@intel.com>

* Replace variable values in example task
Co-Authored-By: mhoffmann1 <marcin.hoffmann@intel.com>

mattclay committed:
Add retries for Invoke-ScriptAnalyzer in pslint.

Hopefully this will work around the intermittent CI failures due
to NullReferenceException, which then succeed on a retry.

acozine committed:
Command module docs: args vs argv (#49907)

* add note about 'args', update examples
* Update lib/ansible/modules/commands/command.py

Co-Authored-By: acozine <879121+acozine@users.noreply.github.com>

bmillemathias committed:
Fix typo in ds_type in examples section (#50166)

replace ds_type elasticisearch by elasticsearch

richlv committed:
Update zabbix_action.py (#50149)

Mostly typo/grammar fixes. "Zabbix Server" was changed to just "Zabbix", as it could be slightly confusing - some people might interpret this referencing the server process, although API matters. "capital-case" was changed, as it usually is one of "capital" or "upper-case".

+label: docsite_pr

samdoran committed:
Fix ansible-runner test

Change test to support runner_on_start in Ansible 2.8 and ansible-runner 1.2.0

Qalthos committed:
Log netconf connection a la paramiko (#50169)

Commit:
optimising vmss tests (#50175)

bincyber committed:
New Module: Add bitbucket_deploy_key (#27467)

* added new module to manage deploy keys for BitBucket repositories
* Modified module to pass CI tests
* Updated bitbucket_deploy_key module
* updated message returned when there is an existing deploy key
* bumped version_added to 2.8
* fixed issues reported by ansible-test

ckotte committed:
VMware: Improve vmware_host_ntp module (#47273)

Akasurde committed:
Fix type in RETURN for bitbucket_deploy_key

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

YouhuaLi committed:
add range_start and range_end agrs when query vm by a single id (#50056)

Commit:
fixing network client version (#50182)

Commit:
sku in the sample became invalid, so updating (#50177)

NilashishC committed:
Fix eos terminal plugin (#49920)

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

justjais committed:
New Module for creating, managing Infoblox NIOS nameserver groups (#49124)

* adding new nios_nsgroup module

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* adding new nios_nsgroup module

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* adding new nios_nsgroup module

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* fix shippable errors

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* added unit tests

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

mdrose committed:
influxdb_user: Fixed unhandled exception on invalid login credentials (#50134)

maiqueb committed:
os_security_group_rule proper module exit (#50076)

When the security group the rule belongs to does not exist and
the state is absent, the module is not properly exited, leading
to a playbook execution failure.

Fixes issue #50057

dbnicholson committed:
Skip non-management Packet IP addresses (#48168)

Non-management (elastic) IP addresses require manual configuration on
the host as described in https://help.packet.net/article/54-elastic-ips.
Skip those so that only the automatically configured management
addresses are used. Otherwise, a non-routable address may be returned in
the inventory.

warusadura committed:
Update `Unknown error` to specific error message (#50129)

* Update `Unknown error` to specific error message

outputs the Exception received rather than just 'Unknown error'
fixes: issue/49713

* Update `Unknown error` to specific error message

improves the error message

hanshell committed:
Update playbooks_intro.rst (#50187)

carchi8py committed:
Rewrite (#49570)

gundalow committed:
Remove incorect author (#50191)

Akasurde committed:
Move doc fragments to changelogs/fragments dir (#50179)

Move changelog fragment to changelogs/fragments dir

richlv committed:
Update ini.py (#50171)

Typo/grammar/consistency fixes.

jonpugh committed:
Change the Packet.net section to be more consistent with the other *Cloud Guide pages. (#50170)

<!--- Your description here -->

+label: docsite_pr

Zempashi committed:
List correctly current PV in "lvg" module: fix lvg reduce  (#49731)

* Refactor integration test for lvg module to introduce grow/reduce test

* List correctly current PV in lvg module: fix lvg reduce

Previous behaviour was to only take into account PV passed in 'pvs'
argument. This lead to reduce not working as expecting:

* with state=present and list of wanted pvs, lvg found only the pvs to
  add or already present and ignored the pv to remove (obviously absent
  from the list of given PV)

* with state=absent and a pv to remove, lvg found that the remaining pvs
  list is empty (ignoring possible other PV in the vg) and decides to
  remove the vg entirely (as supposely no PV are left anymore to store
  lvm metadata)

* Add changelog fragment

fatmcgav committed:
Jenkins_plugin: Handle 'latest' version when installing plugin for first time (#49723)

* Handle 'latest' version when installing module for first time

Otherwise, a fresh install of a plugin with 'version: latest' gets installed without its dependencies, rendering the plugin effectively useless.

* Add changelog

felixfontein committed:
Prevent error if module.params['path'] has no path included. (#50106)

dagwieers committed:
MSC: Various bugfixes and features enhancements (#50200)

* MSC: Various bugfixes and features enhancements

This PR includes:
- Lookups of roles, labels and domains
- Auto-create new labels
- Improvements to comparing complex datastructures
- Force removal of sites
- Support non top-level queries
- Document internal functions
- Add parameter types to modules
- Fix documentation examples
- Improvements to idempotency wrt. returning changed
- Support site locations
- Update permission list
- Various improvements to integration tests

* Fix Ci issues

dagwieers committed:
Docs: Show parameter types (in purple) (#49966)

* Docs: Show parameter types (in purple)

* Changes based on feedback

* Remove leftover statement after review

* Simplify TOC and support section

* Add missing 'v' to version_added

* Remove the v for version

* Update docs/templates/plugin.rst.j2

Co-Authored-By: dagwieers <dag@wieers.com>

* Update docs/templates/plugin.rst.j2

Co-Authored-By: dagwieers <dag@wieers.com>

* Move Author into Support section

* Avoid more "isn't included in any toctree" errors

* Add Red Hat support section, list module status

cognifloyd committed:
Make modprobe module check for builtins as well (#37150)

Without this modprobe always reports changed when modprobe-ing a builtin module.

With this, if a kernel module is a builtin, the modprobe module will:
- succeed (without incorrectly reporting changed) if ``state`` is ``present``;
- fail if ``state`` is ``absent``

The failure will have whatever error message modprobe returns when
attempting to remove a builtin module. For example:
``modprobe: ERROR: Module nfs is builtin.``

Qalthos committed:
Fix some handling of log messages from ansible-connection (#50214)

zharalim committed:
Add more detailed documentation on how to use multiple inventories (#47586)

* Add a new section on how to use multiple inventory sources w/ examples

Co-Authored-By: zharalim <zharalim@outlook.com>

Akasurde committed:
Misc documentation update (#50190)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

jborean93 committed:
Added supershipit to botmeta sanity test (#50219)

rambleraptor committed:
Giving Google engineers supershipit on GCP Ansible resources/plugins/tests (#50218)

dagwieers committed:
win_updates: Correctly report changes on success (#50188)

* win_updates: Correctly report changes on success

* Add changelog fragment

dagwieers committed:
win_firewall_rule: Remove invalid 'bypass' action (#50207)

* win_firewall_rule: Remove invalid 'bypass' action

This fixes #50143

* Add changelog fragment

plastikat committed:
Fix docs typo at Special Variables (#50230)

mattclay committed:
Add `env` command to ansible-test and run in CI. (#50176)

* Add `env` command to ansible-test and run in CI.
* Avoid unnecessary docker pull.

anasbadaha committed:
Fix Issue #49612 (#50073)

* Fix Issue #49612

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Fix Pep8 failures

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Add validation for send empty group name

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Fix Error message

Signed-off-by: Anas Badaha <anasb@mellanox.com>

anasbadaha committed:
Fix Issue #39598 (#49982)

* Fix Issue #39598

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Fix unit test for onyx_config

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Add comments for save running config

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Enhance onyx_config bug fix

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Fix result['changed'] = True comment

Signed-off-by: Anas Badaha <anasb@mellanox.com>

Tomorrow9 committed:
VMware: add VBS configuration support for Windows 10 (#48595)

* Fix space issue
* fix whitespaces issue

ckotte committed:
Revert changes in vmware_vcenter_statistics(#49537)

Akasurde committed:
postgresql_idx: misc fixes (#50199)

* Removed unused import
* Removed 'login_password' from exit_json call
* Removed unused db, cond variable

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
Correct match test documentation (#50229)

Match test matches zero or more characters at the begining of the string.

Fixes: #32366

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Commit:
fixed invalid samples (#50227)

* fixed invalid samples

Qalthos committed:
Add backup parameter to cli_config (#50206)

* Add backup parameter to cli_config

* Add a unit test for cli_config backup

Qalthos committed:
Merge backup functionality across config action plugins (#50208)

* Merge backup functionality across config action plugins

* Port updated cli_config

Qalthos committed:
Log device interaction and push labeled logs back to controller (#50028)

* Add session tracing support for network_cli, netconf and httapi connection

*  Add `persistent_log_messages` configuration option to log device inteaction
   in log file for network_cli, netconf and httapi connection
   type
*  Log jsonrpc request and response in log file is configuration option
   is enabled

* Update docs to talk about warning shown when persistent_log_messages is on

rambleraptor committed:
New GCP Module: gcp_sql_user_facts (#48273)

Qalthos committed:
Allow AnsibleVaultEncryptedUnicode to pass through exec_jsonrpc (#48306)

* Allow AnsibleVaultEncryptedUnicode to pass through exec_jsonrpc

* Add changelog

rambleraptor committed:
New GCP Module: gcp_sql_instance_facts (#48271)

rambleraptor committed:
GCP Bug Fixes (#49458)

* Bug fixes

* fixes as of 12/5/2018

* changes as of 12/06/2018

rambleraptor committed:
New GCP Module: gcp_sql_database_facts (#48269)

rambleraptor committed:
New GCP Module: gcp_spanner_instance_facts (#48267)

rambleraptor committed:
New GCP Module: gcp_spanner_database_facts (#48265)

rambleraptor committed:
New GCP Module: gcp_resourcemanager_project_facts (#48263)

rambleraptor committed:
New GCP Module: gcp_iam_service_account_facts (#48257)

rambleraptor committed:
New GCP Module: gcp_container_node_pool_facts (#48251)

rambleraptor committed:
New GCP Module: gcp_container_cluster_facts (#48249)

rambleraptor committed:
New GCP Module: gcp_compute_interconnect_attachment_facts (#48217)

gitaaron committed:
ansible-doc: update faq to include common z/OS pitfalls (#18555) (#49504)

* ansible-doc: update faq to include common z/OS pitfalls (#18555)
Co-Authored-By: gitaaron <aaron.surty@gmail.com>

dagwieers committed:
Fix typo

glitchcrab committed:
update github handle (#50235)

lingfish committed:
Update nios_aaaa_record.py (#50271)

Should be ipv6.

+label: docsite_pr

Akasurde committed:
postgresql_user: set type of conn_limit (#50178)

conn_limit type is set to 'int'. This will allow module to compare conn_limit with record value without type casting.

Fixes: #38118

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

curry9999 committed:
Add colon after module name (#50274)

* A colon is not described

* dddd

* Update aws_kms_facts.py

kvaps committed:
fix: ansible_kubectl_ssl_ca_cert variable (#50217)

* fix: mistake for ansible_kubectl_ssl_ca_cert variable

* fix: K8S_AUTH_VERIFY_SSL

felixfontein committed:
Elliptic curve tests for crypto modules (#50109)

* Add openssl_csr ECC test.

* Add openssl_publickey ECC test.

* Add openssl_certificate ECC test.

gundalow committed:
Revert TEMP HACK: Upload all to codecov (#50279)

Codecov.io have made some improvements to results processing.
Upload all the coverage data to see is codecov can process this.

Reverts https://github.com/ansible/ansible/pull/47515

ganeshrn committed:
Fix for AttributeError in network_cli (#50284)

*  Value of prompt_answer is wrongly used as `self.prompt_answer`
   while logging it in message queue

rajaspachipulusu17 committed:
Pluribus Networks igmp snooping module with Unit tests (#50146)

rubentsirunyan committed:
Zabbix Action: Idempotency fix (#50297)

NilashishC committed:
Change net_* default behaviour on missing module (#49923)

* Fix net_base default behaviour

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

* New fix

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

ushuz committed:
docker_image: add network parameter (#50313)

* docker_image: add network parameter

Add a network parameter to the docker_image module to specify the
network to use for RUN commands.

Called it network instead of network_mode as the latter seems like a
legacy of when there were just a few default options to choose from,
while now the name of an arbitrary network can be specified.

Fixes #21433

* Format description

* Update docker_image option_minimal_versions

* Ensure network_mode param optional

maxamillion committed:
Provide both service state and status when possible in service_facts (#49618)

* Combine systemd units/unit-files output for service_facts

Fixes #47118

Previously we were only taking the output of `systemd units` which
would leave out information about the service units that were
disabled, static, masked, etc. Now we're aggregating the results so
that anything not active/inactive/dead at least is pulled as fact
data with it's state provided.

Signed-off-by: Adam Miller <admiller@redhat.com>

* provide state and status information about services

Fixes #45730

Signed-off-by: Adam Miller <admiller@redhat.com>

felixfontein committed:
ACME: add seealso references (#50320)

* Add seealso references to ACME modules.

* Bump to latest drafts.

felixfontein committed:
Improve tagging examples. (#50321)

sdodsley committed:
Fix bad placement of array variable (#50256)

curry9999 committed:
Documentation remove escape r (#50277)

* A colon is not described

* dddd

* Update branch

gundalow committed:
Debug: Size of coverage file (#50380)

felixfontein committed:
ufw: add integration tests (#50374)

* Add basic integration tests for ufw.

* Being more honest.

dagwieers committed:
Add missing aliases file (#50357)

gundalow committed:
Disable mysql_db (#50385)

Akasurde committed:
Temporarily disable mysql_user testcases (#50395)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
Temporarily disable mysql_variables (#50403)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

sdodsley committed:
Pure Storage facts fix (#50349)

* Add module for Pure Storage FlashBlade to manage directory services

* Fix facts not correctly passing into ansible_facts dict

Commit:
apply jborean93 requested changes

Commit:
apply jborean93 requested changes

Pat-Hof commented:
Hi,

> It would be great if you could add some integration tests, feel free to reach out if you are unsure on how to do this.

Where can I find  some documentation/hints.example for writing this tests ? I've never done something before.

Best Regards

Commit:
Merge branch 'win_package' of https://github.com/Pat-Hof/ansible into win_package

mattclay commented:
Closing to avoid polluting the [by file report](https://ansible.sivel.net/pr/byfile.html).

Feel free to open a new PR, comment here to ask for help, or ask on #ansible-devel on Freenode about rebasing a PR. The [rebasing documentation](https://docs.ansible.com/ansible/devel/dev_guide/developing_rebasing.html) may also be helpful.
